core/src/subgraph/context/instance.rs

```diff
@@ -155,7 +155,35 @@ where
         })
     }
 
-    pub(super) fn revert_data_sources(&mut self, reverted_block: BlockNumber) {
+    pub(super) fn revert_data_sources(
+        &mut self,
+        reverted_block: BlockNumber,
+    ) -> Vec<DataSource<C>> {
```

Comment:
Considering we don't need to return removed on-chain data sources, this could return `Vec<offchain::DataSource>`. We could even return just `Vec<offchain::Source>` to minimize cloning.